deploy/docker-swarm/pkg/Dockerfile

```diff
@@ -1,12 +1,12 @@
-FROM golang:1.19 as builder
+FROM --platform=$TARGETPLATFORM golang:1.19 as builder
 WORKDIR /csi
 ADD go.mod go.sum /csi/
 RUN go mod download
 ADD . /csi/
 RUN ls -al
-RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o aio.bin github.com/hetznercloud/csi-driver/cmd/aio
+RUN GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -o aio.bin github.com/hetznercloud/csi-driver/cmd/aio
```

Comment:
This will always build `arm64` images. I think if you run with `--platform=$TARGETPLATFORM` you do not need to specify `GOOS` & `GOARCH` as they will default to the current os/arch the image is built on.